authors/apps/ratings/views.py

```diff
@@ -0,0 +1,94 @@
+from django.db.models import Avg
+from rest_framework import status
+from rest_framework.exceptions import NotFound
+from rest_framework.generics import (
+    GenericAPIView
+)
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly
+)
+from rest_framework.response import Response
+
+from .models import Article, Rating
+from .serializers import RatingSerializer
+
+
+class ArticleRatingAPIView(GenericAPIView):
+    queryset = Rating.objects.all()
+    serializer_class = RatingSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def get_article(self, slug):
+        """Returns the first record in the articles table with the slug"""
+        article = Article.objects.all().filter(slug=slug).first()
+        return article
+
+    def post(self, request, slug):
+        """POST Request to rate an article"""
+        rating = request.data
+        article = self.get_article(slug)
+
+        if not article:
+            return Response({
+                "message": "Article not found"
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if request.user.id == article.author.id:
+            return Response({
+                "message": "You cannot rate your own article"
+            }, status=status.HTTP_403_FORBIDDEN)
+        try:
+            current_article_rating = Rating.objects.get(
+                user=request.user.id,
+                article=article.id
+            )
+            serializer = self.serializer_class(
+                current_article_rating, data=rating)
+        except Rating.DoesNotExist:
+            serializer = self.serializer_class(data=rating)
+
+        serializer.is_valid(raise_exception=True)
+        serializer.save(user=request.user, article=article)
+        return Response({
+            'message': 'Rating submitted successfully!',
+            'data': serializer.data
+        }, status=status.HTTP_201_CREATED)
+
+    def get(self, request, slug):
+        """Returns an article's ratings"""
+        article = self.get_article(slug)
+        rating = None
+
+        if not article:
+            return Response({
+                "message": "Article not found"
+            }, status=status.HTTP_404_NOT_FOUND)
+        if request.user.is_authenticated:
+            try:
+                rating = Rating.objects.get(
+                    user=request.user, article=article
+                )
+            except Rating.DoesNotExist:
+                raise NotFound(
+                    detail={'error': 'You have not yet rated this article'}
+                )
+
+        if rating is None:
+            avg = Rating.objects.filter(
```

Comment:
This functionality exists in the serializer class defined for this view. Have you considered using that?